### PR TITLE
feat(card): hadnle add card flow via setup intent

### PR DESCRIPTION
**Pull request by OlegDO**
Tests will be in another PR

### CODE REVIEW

File: microservices/payment/src/entities/card.ts [hunk 1]
```diff
@@ -1,6 +1,92 @@
...
+    description: 'Field for storing id of according card entity',
+    example: 'card_1N7zzsFpQjUWTpHeCrYKrCkR',
+  })
+  @PrimaryColumn({ type: 'varchar', length: 19 })
+  @Length(1, 29)
+  cardId: string;
+
+  @Index('IDX_card_userId', ['userId'])
+  @Column({ type: 'varchar', length: 36 })
+  @Unique(['userId'])
```
Comment (MatthewPattell) [inline microservices/payment/src/entities/card.ts]: you can have multiple cards

File: microservices/payment/src/entities/card.ts [hunk 2]
```diff
@@ -1,6 +1,92 @@
...
+})
 @Entity()
-class Card {}
+class Card {
+  @JSONSchema({
+    description: 'Field for storing id of according card entity',
+    example: 'card_1N7zzsFpQjUWTpHeCrYKrCkR',
+  })
+  @PrimaryColumn({ type: 'varchar', length: 19 })
+  @Length(1, 29)
```
Comment (MatthewPattell) [inline microservices/payment/src/entities/card.ts]: compare length for validator and typeorm

File: microservices/payment/src/entities/card.ts [hunk 3]
```diff
@@ -1,6 +1,92 @@
...
+  @Column({ type: 'varchar', length: 36 })
+  @Unique(['userId'])
+  @Length(1, 36)
+  userId: string;
+
+  @JSONSchema({
+    description: 'Last 4 card digits',
+    example: '4242',
+  })
+  @Column({ type: 'varchar' })
```
Comment (MatthewPattell) [inline microservices/payment/src/entities/card.ts]: length?

File: microservices/payment/src/entities/card.ts [hunk 4]
```diff
@@ -1,6 +1,92 @@
...
+  @JSONSchema({})
+  @Column({ type: 'varchar', length: 255, default: '' })
+  @IsUndefinable()
+  @Length(1, 255)
+  holderName: string;
+
+  @JSONSchema({
+    example: 'visa',
+  })
+  @Column({ type: 'varchar', length: 20 })
```
Comment (MatthewPattell) [inline microservices/payment/src/entities/card.ts:66]: validator?

File: microservices/payment/src/entities/card.ts [hunk 5]
```diff
@@ -1,6 +1,92 @@
...
+  })
+  @Column({ type: 'varchar', length: 5 })
+  @IsCardExpirationValid()
+  @Length(5, 5)
+  expired: string;
+
+  @JSONSchema({})
+  @Column({ type: 'varchar', length: 255, default: '' })
+  @IsUndefinable()
+  @Length(1, 255)
```
Comment (MatthewPattell) [inline microservices/payment/src/entities/card.ts]: I think it's can be shorter - 100

File: microservices/payment/src/entities/card.ts [hunk 6]
```diff
@@ -1,6 +1,92 @@
...
+    example: 'visa',
+  })
+  @Column({ type: 'varchar', length: 20 })
+  type: string;
+
+  @JSONSchema({
+    description: "If it's the first attached user card it should be default",
+  })
+  @Column({ type: 'boolean', default: false })
+  @IsBoolean()
```
Comment (MatthewPattell) [inline microservices/payment/src/entities/card.ts:75]: undefinable

Comment (MatthewPattell) [reply microservices/payment/src/entities/card.ts:75]: for pass partial field

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/payment/src/entities/card.ts (hunk 4 above)
Comment (OlegDO) [reply microservices/payment/src/entities/card.ts:66]: added

File: microservices/payment/src/entities/card.ts (hunk 6 above)
Comment (OlegDO) [reply microservices/payment/src/entities/card.ts:75]: added

File: microservices/payment/src/services/payment-gateway/stripe.ts [hunk 7]
```diff
@@ -262,6 +266,9 @@ class Stripe extends Abstract {
         case 'checkout.session.completed':
           void this.handleTransactionCompleted(event);
           break;
+        case 'setup_intent.succeeded':
```
Comment (MatthewPattell) [inline microservices/payment/src/services/payment-gateway/stripe.ts]:
```suggestion

        case 'setup_intent.succeeded':
```

**Review by MatthewPattell — APPROVED** [review]

**Pull request merged**